homeassistant/components/light/homematicip_cloud.py

```diff
@@ -79,3 +82,38 @@ def device_state_attributes(self):
             ATTR_ENERGIE_COUNTER: round(self._device.energyCounter, 2)
         })
         return attr
+
+
+class HomematicipDimmer(HomematicipGenericDevice, Light):
+    """MomematicIP dimmer light device."""
+
+    def __init__(self, home, device):
+        """Initialize the dimmer light device."""
+        super().__init__(home, device)
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._device.dimLevel != 0
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self._device.dimLevel*255.0
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            await self._device.set_dim_level(
+                float(kwargs[ATTR_BRIGHTNESS])/255.0)
```

Comment:
The copy to float shouldn't be necessary. Float division is the default in Python 3. The brightness argument is already validated as an integer by the service schema.